### PR TITLE
fix error with collectionfilter=1 not being included in batch links

### PR DESCRIPTION
**Pull request by djay**

### CODE REVIEW

File: src/collective/collectionfilter/testing.py [hunk 1]
```diff
@@ -60,6 +60,13 @@ def setUpPloneSite(self, portal):
                 text=RichTextValue(u'Ein heißes Test Dokument'),
                 subject=[u'Süper', u'Dokumänt'],
             )
+            portal.invokeFactory(
+                'Document',
+                id='testdoc2',
+                title=u'Page 😉',
```
Comment (thet) [inline src/collective/collectionfilter/testing.py]: :)

File: src/collective/collectionfilter/testing.py [hunk 2]
```diff
@@ -60,6 +60,13 @@ def setUpPloneSite(self, portal):
...
+                text=RichTextValue(u'Ein heißes Test Dokument'),
```
Comment (thet) [inline src/collective/collectionfilter/testing.py]: :thumbsup:

File: src/collective/collectionfilter/contentfilter.py [hunk 3]
```diff
@@ -9,6 +9,6 @@ def set_content_filter(context, event):
     req = event.request
     if 'collectionfilter' not in req.form:
         return
-    del req.form['collectionfilter']
+    # We leave collectionfilter=1 in req.form so that it gets put into batch links
```
Comment (thet) [inline src/collective/collectionfilter/contentfilter.py]: The reason for removing the ``collectionfilter`` url parameter was to exit early and not have to ``make_query`` for each traversal step - which can be a lot.
Maybe we should instead check for the existance of ``event.request['contentFilter']`` - if possible?

**Comment by djay** [comment]
Ok but I wasn't sure there was multiple traversals?

**Comment by thet** [comment]
@djay I'd have to step in myself and see how often this is typically called. There is a chance that this is only a micro optimization but this code is called for all path-parts (IIRC).

**Comment by djay** [comment]
@thet I added the check but not sure what you all path parts you mean. its still skipping if collectionfilter=1 isn't in there so it would only other parts of collectionfilter traversals that would call make_query.

Also, any idea why 5.2 is failing? I don't think it's something I did

**Comment by djay** [comment]
@thet Given this is a major bug and breaks batching can we get this merged and then worry about optimisations later?

**Comment by petschki** [comment]
@djay can u resolve the conflicts? I'll merge this one if tests are green.

**Comment by djay** [comment]
@petschki failing due to merging my reversion that means the rebase also removes the code :(
One reason I think it's better to do a squash then merge to close a PR. What goes on in a PR stays in a PR!

**Comment by petschki** [comment]
@djay I've fixed another missing code snippet due to your rebase. Local tests are now green. I wait for travis and merge later.

**Pull request merged**